modules/service/data-service.js

```diff
@@ -328,7 +328,13 @@ class DataService {
             //     result = this.data.execute(query);
             //     break;
             case 'CONSTRUCT':
-                result = this.data.construct(query);
+                result = await this.data.construct(query);
+                result = result.toString();
```

Comment:
Maybe t move this inside if, so only if we have result to call toString

Reply:
What happens if construct returns us bufferList without anything inside and we call toString?

Reply:
I have already tested it with query that returns empty response and it worked well

Reply:
I guess that if we have empty bufferList it will pass the if statement. Let's add this to the parking lot since we have to refactor it definitely

Reply:
Understand, cool